gpiozero/boards.py

```diff
@@ -1185,3 +1185,86 @@ def on(self):
     def off(self):
         self.value = False
 
+class SevenSegmentDisplay(LEDBoard):
+    layouts = {
+        '1': (False, True, True, False, False, False, False),
+        '2': (True, True, False, True, True, False, True),
+        '3': (True, True, True, True, False, False, True),
+        '4': (False, True, True, False, False, True, True),
+        '5': (True, False, True, True, False, True, True),
+        '6': (True, False, True, True, True, True, True),
+        '7': (True, True, True, False, False, False, False),
+        '8': (True, True, True, True, True, True, True),
+        '9': (True, True, True, True, False, True, True),
+        '0': (True, True, True, True, True, True, False),
+        'A': (True, True, True, False, True, True, True),
+        'B': (False, False, True, True, True, True, True),
+        'C': (True, False, False, True, True, True, False),
+        'D': (False, True, True, True, True, False, True),
+        'E': (True, False, False, True, True, True, True),
+        'F': (True, False, False, False, True, True, True),
+        'G': (True, False, True, True, True, True, False),
+        'H': (False, True, True, False, True, True, True),
+        'I': (False, False, False, False, True, True, False),
+        'J': (False, True, True, True, True, False, False),
+        'K': (True, False, True, False, True, True, True),
+        'L': (False, False, False, True, True, True, False),
+        'M': (True, False, True, False, True, False, False),
+        'N': (True, True, True, False, True, True, False),
+        'O': (True, True, True, True, True, True, False),
+        'P': (True, True, False, False, True, True, True),
+        'Q': (True, True, False, True, False, True, True),
+        'R': (True, True, False, False, True, True, False),
+        'S': (True, False, True, True, False, True, True),
+        'T': (False, False, False, True, True, True, True),
+        'U': (False, True, True, True, True, True, False),
+        'V': (False, True, True, True, True, True, False),
+        'W': (False, True, False, True, False, True, False),
+        'X': (False, True, True, False, True, True, True),
+        'Y': (False, True, True, True, False, True, True),
+        'Z': (True, True, False, True, True, False, True),
+        '-': (False, False, False, False, False, False, True),
+        ' ': (False, False, False, False, False, False, False)
+        }
+    
+    def __init__(self, *pins, **kwargs):
+        # 7 segment displays must have 7 or 8 pins
+        if len(pins) < 7 or len(pins) > 8:
+            raise ValueError('7 segment display must have 7 or 8 pins')
+        # Don't allow 7 segments to contain collections
+        for pin in pins:
+            assert not isinstance(pin, LEDCollection)
+        pwm = kwargs.pop('pwm', False)
+        active_high = kwargs.pop('active_high', True)
+        if kwargs:
```

Comment:
In common with LEDBoard, should we also have a boolean `initial_value` property?